hudi-utilities/src/test/java/org/apache/hudi/utilities/deltastreamer/TestHoodieDeltaStreamer.java

```diff
@@ -1521,10 +1656,16 @@ public void testFilterDupes(HoodieRecordType recordType) throws Exception {
     // Try UPSERT with filterDupes true. Expect exception
     cfg2.filterDupes = true;
     cfg2.operation = WriteOperationType.UPSERT;
+    HoodieDeltaStreamer ds4 = null;
     try {
-      new HoodieDeltaStreamer(cfg2, jsc).sync();
+      ds4 = new HoodieDeltaStreamer(cfg2, jsc);
```

Comment:
we can move this initialization outside try block. Then null check will not be needed in finally block, as we are doing at other places.

Reply:
Updated